src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -3102,10 +3102,8 @@ static int add_match_callback(
                 r = 1;
         }
 
-        if (failed && match_slot->floating) {
+        if (failed && match_slot->floating)
                 bus_slot_disconnect(match_slot);
-                sd_bus_slot_unref(match_slot);
```

Comment:
this fix isn't right. Coverity is wrong here... when the slot is "floating" it has an extra ref, and we need to get rid of it here...

(The concept of "floating" means, the "sd_bus_slot" ptr was NULL when the match was registered and hence the matche's lifetime is bound to the lifetime of the sd_bus object it belongs to. In that case we keep an extra ref on it, which makes sure the match survives as long as needed)

Reply:
Yeah, I was just looking at that, and it seems that the sd_bus_slot unref at the end matches the ref at the beginning, and no other ref changes occur in between.

Reply:
Oh, indeed. I will drop this. Thanks.